### PR TITLE
Add GCS e2e tests

### LINKED ISSUE #373: Add e2e tests for GCS
This particularly needs a new image with the GCS storage.

### PR DESCRIPTION
**Pull request by ivanyu**
Closes #373

### CODE REVIEW

File: .github/workflows/main_push_and_pull_request_workflow.yml [hunk 1]
```diff
@@ -35,7 +35,7 @@ jobs:
     strategy:
       matrix:
         java-version: [ 11 ]
-        test: [ 'LocalSystem', 'S3' ]
+        test: [ 'LocalSystem', 'S3', 'GCS' ]
```
Comment (jeqo) [inline .github/workflows/main_push_and_pull_request_workflow.yml:38]: Locally failed with `GCS` "tests not found", maybe safer to use `Gcs`?

Comment (ivanyu) [reply .github/workflows/main_push_and_pull_request_workflow.yml:38]: The problem is worse...
Look at the "S3" E2E run
image

I'll fix this separately.

Comment (ivanyu) [reply .github/workflows/main_push_and_pull_request_workflow.yml:38]: Here it is: https://github.com/Aiven-Open/tiered-storage-for-apache-kafka/pull/412

Comment (jeqo) [reply .github/workflows/main_push_and_pull_request_workflow.yml]: 🤦🏽

Comment (ivanyu) [reply .github/workflows/main_push_and_pull_request_workflow.yml]: Should be good now

**Review by jeqo — APPROVED** [review]
Thanks @ivanyu !

**Pull request merged**